pkg/controllers/osc/osc_controller.go

```diff
@@ -49,6 +50,8 @@ const (
 	MachineDeploymentCleanupFinalizer = "kubermatic.io/cleanup-operating-system-configs"
 	// CloudInitSettingsNamespace is the namespace in which OSCs and secrets are created by OSC controller
 	CloudInitSettingsNamespace = "cloud-init-settings"
+	// MachineDeploymentChecksum is the checksum generated against MachineDeployment Spec
+	MachineDeploymentChecksum = "machine-deployment-checksum"
```

Comment:
Why not using the machine deployment version instead? annotated this in the OSC when the machine deployment is created, and check agains the updated machine deployment version.

Reply:
That was one option but I just wanted a resource(annotation) that we manage. We can use `machinedeployment.clusters.k8s.io/revision` for this as well.

Reply:
But this exactly why the version is there. Why should we build something from scratch and use it instead of just simply use the resource version, which exactly what this field is for ;)

Reply:
Updated.